clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -324,6 +324,8 @@ def warn_drv_treating_input_as_cxx : Warning<
   InGroup<Deprecated>;
 def warn_drv_pch_not_first_include : Warning<
   "precompiled header '%0' was ignored because '%1' is not first '-include'">;
+def warn_drv_existing_archive_append: Warning<
+  "appending to an existing archive '%0'">, InGroup<DiagGroup<"archive-append">>;
```

Comment:
is the `InGroup` specifier necessary?  There are no other diagnostics in the group.

Reply:
The `InGroup` is necessary to expose some sort of a `-Wno` flag for the warning (which is necessary for all new warnings).
My impression has been that whenever some warning can't be put into a particular group, this `DiagGroup<custom-warning-type>` pattern is commonly used to expose `-Wno-custom-warning-type`.
Thus, `-Wno-archive-append` is provided for this message.